connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/HerderRequestHandler.java

```diff
@@ -41,18 +41,11 @@ public class HerderRequestHandler {
 
     private final RestClient restClient;
 
-    private volatile long requestTimeoutMs;
+    private final RestRequestTimeout requestTimeout;
```

Comment:
The field `requestTimeoutMs` was made volatile as part of [here](https://github.com/apache/kafka/pull/14562/files#diff-f2311f0c356f882d7768b97cb5f0054dc7040d29d455eab74ab585725454488aR44). I don't think we need it now, but just wanted to understand why was it added over there (you can skip the explanation if it's too complex :) )

Reply:
This field is final, it wouldn't make sense to mark it volatile.

The underlying mutable value that it returns from `timeoutMs()` is still marked volatile in the only non-test implementation [here](https://github.com/C0urante/kafka/blob/0713736b9884084c360126558e4b6d92d4435bfa/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestServer.java#L533).

Reply:
I didn't intend to mark the final field as volatile. I wanted to know why the field was marked as volatile in the old PR since you had made the change.

Reply:
Ah, gotcha. The intent was to make it thread-safe since it's likely that writes and reads to/of that value will take place on different threads.